### DISCUSSION TITLE
Avoid repeat navigation instructions

### DISCUSSION THREAD

**Discussion by RHoffi** [devemux86/cruiser#146] [Q&A] [answered]
When driving on a Highway on every street, i get 3 Notifications to follow the street. Is it possible to disable these? They are not helpful.

**Comment by devemux86** [comment]
Hi @RHoffi,

Do you mean the voice instructions "In X meters turn right"?

You can turn on / off the navigation instructions you want in
"Settings | Voice guidance | Advanced | Navigation instructions" or search settings for "instructions".

You can also turn on / off the "Settings | Voice guidance | Advanced | Repeat navigation instructions".

Please also see the images here:
- https://github.com/devemux86/cruiser/discussions/108#discussion-5022326

**Reply by RHoffi** [reply]
The Notification is in x Meters follow the street. This happens on Highways on an exit. Turn right or left makes sense, if I had to leave the highway.

**Comment by devemux86** [comment]
@RHoffi 

Do you mean that a turn instruction somewhere is wrong?
The turn instructions are provided by the routing services and can be seen in menu "Routing | Turn instructions".

Can you post a route example?
What routing service do you use?

**Reply by RHoffi** [reply]
https://kurv.gr/xTkAU
Between Shaping Point 4 and 5,3 Notifications  on the Highway for every exit. I use Kurviger Web for planning, but I prefer Cruiser for Navigation.

**Reply by RHoffi** [reply]
Instructions are: follow the Road for x kilometers

**Reply by devemux86** [reply]
@RHoffi thanks for using Cruiser and the example!

Between shaping points 4 and 5 there are 2 turn instructions (that should be announced):
- Keep right
- Turn left

Can you write more details what do you mean with 3 notifications?
What exactly does voice guidance speak?

Also what options are enabled in "Settings | Voice guidance | Advanced"?
Can you post a screenshot of settings or export the app .pref file via "Settings | Application | Advanced | Preferences"?

![Kurviger](https://user-images.githubusercontent.com/3484020/235352607-75f44a25-ec54-468e-b2b8-8d5dd4a4a445.jpg)

**Reply by devemux86** [reply, marked as answer]
> Instructions are: follow the Road for x kilometers

Such announcement at regular intervals can happen when the next turn instruction is too far away.

This can be turned on / off in "Settings | Voice guidance | Advanced | Repeat navigation instructions".
Do you have it enabled?

You can also change the distance in "Settings | Voice guidance | Distances | Profile | Repeat navigation instructions".
What distance is set?

**Reply by RHoffi** [reply]
Thanks for your help, I will try your suggestions on my next trip. Repeat navigation instructions was on and distance was set to 4km. Will turn it off.

**Reply by devemux86** [reply]
According to the turn instructions list (seen [above](https://github.com/devemux86/cruiser/discussions/146#discussioncomment-5765937)), the distance between the shaping point 4 and the next turn instruction is 126 km.

That's why there were repeated navigation instructions at regular intervals (if the setting is enabled).
You could also increase the "repeat" distance, if you want such instructions.

Currently the max distance can be set 10 km, perhaps we need a larger maximum?

**Reply by RHoffi** [reply]
A larger Maximum would bei nice. Again thanks for your help

**Reply by devemux86** [reply]
I will increase the max distance option for the repeated navigation instructions in the next Cruiser update.
